src/components/search/SearchPanel.tsx

```diff
@@ -0,0 +1,233 @@
+    const options = tagNames.filter((tag) =>
+      tag.toLowerCase().includes(searchText.slice(1).toLowerCase())
+    );
+    const selectedValues = searchResults
+      .filter((result) => result.type === searchMode)
+      .map((result) => result.value);
+    return options.filter((option) => !selectedValues.includes(option)).map((e) => "#" + e);
+  };
+
+  const handleInputValueChange = (value: string) => {
+    setErrMsg("");
+    setSearchText(value);
+    if (value.startsWith("@")) {
+      setSearchMode("username");
+    } else if (value.startsWith("#")) {
+      setSearchMode("tag");
+    } else {
+      setSearchMode("title");
+    }
+  };
+
+  return (
+    <Stack alignItems="center" direction="column">
+      <Autocomplete
+        sx={{ width: "500px" }}
+        disableClearable
+        freeSolo
+        inputValue={searchText}
+        value={searchText}
+        onInputChange={(_, value) => {
+          handleInputValueChange(value);
+        }}
+        onChange={(_, value) => {
+          if (!value) return;
+          handleSubmit(value);
+        }}
+        options={getOptions()}
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            name="search"
+            placeholder="Search: @username, #tag, title"
+            variant="outlined"
+            error={errMsg != ""}
+            InputProps={{
+              ref: params.InputProps.ref,
+              sx: textfield_overwrite,
+              endAdornment: (
+                <Button
+                  sx={{
+                    border: "3px solid black",
+                  }}
+                  disabled={searchMode == "tag"}
+                  variant="contained"
+                  type="submit"
+                  style={submit_layout}
+                  onClick={() => handleSubmit(searchText)}
+                >
+                  <SearchIcon />
+                </Button>
+              ),
+            }}
+          />
+        )}
+      />
+      <Box sx={helperText}>
+        <FormHelperText error>{errMsg != "" && errMsg}</FormHelperText>
+      </Box>
+      <Box
+        sx={{
+          display: "flex",
+          justifyContent: "center",
+          flexWrap: "wrap",
+          gap: 1,
+          marginTop: "20px",
+          width: "700px",
+        }}
+      >
+        {searchResults.map((result, index) => (
+          <Chip
+            key={index}
+            label={
+              <Stack direction="row" spacing="10px">
+                <Typography variant="body2">
+                  {result.type === "username" ? "HOST:" : result.type === "tag" ? "TAG:" : "TITLE:"}
+                </Typography>
+                <Typography variant="body2" color="primary">
+                  {result.value}
+                </Typography>
+              </Stack>
+            }
+            deleteIcon={<CloseIcon />}
+            onDelete={() => handleDeleteTag(index)}
+          />
```

Comment:
![image](https://user-images.githubusercontent.com/69895436/228532899-8019a7b5-4302-4425-9b5b-169ea6739e38.png)

Tags don't go the same way as design
![image](https://user-images.githubusercontent.com/69895436/228532731-3b2ff453-5214-45d7-bcd9-cdbb1f7f36b4.png)

Reply:
Done

Reply:
![image](https://user-images.githubusercontent.com/69895436/228572037-2929569c-80bd-4dc2-9192-7a0fbabc8570.png)
close icon still grey and in design the border look lighter then the above textfield

Reply:
by lighter, do you mean color or border width

Reply:
The width, my bad

Reply:
done